src/coreComponents/mesh/MeshLevel.cpp

```diff
@@ -263,18 +261,7 @@ MeshLevel::MeshLevel( string const & name,
 
 
 MeshLevel::~MeshLevel()
-{
-  if( !m_isShallowCopy )
-  {
-    delete m_nodeManager;
-    delete m_edgeManager;
```

Comment:
`m_particleManager` wasn't being deleted. I changed these members to `std::shared_ptr` so this won't be a problem anymore.

Reply:
why a `shared_ptr` instead of a `unique_ptr`?

Reply:
nice!

Reply:
As far as I'm aware we're not using `std::shared_ptr` anywhere else and I haven't used it personally much either. But this seems like a prime use case.